### PR TITLE
Fix invalid return code on partial compile error

### PR DESCRIPTION
**Pull request by dansimau**
This better be the last time I do this.

@uber/ringpop

### CODE REVIEW

File: test/go-test-prettify [hunk 1]
```diff
@@ -1,74 +1,104 @@
...
+					echo "$line"
+				fi
+				;;
+
+			# Skip and ignore printing junk
+			PASS|FAIL|\?*)
+				;;
+
+			*)
+				# Output unknown lines
```
Comment (Motiejus) [inline test/go-test-prettify]: Maybe you meant the opposite, i.e. output unknown lines _only_ in verbose mode?

Comment (Motiejus) [inline test/go-test-prettify]: Sorry, I missed that you're outputting them below.

**Comment by Motiejus** [comment]
LGTM

**Comment by thanodnl** [comment]
lgtm

File: test/go-test-prettify (hunk 1 above)
Comment (dansimau) [inline test/go-test-prettify]: :+1:

**Pull request merged**